packages/nuxt/src/core/nuxt.ts

```diff
@@ -408,11 +408,14 @@ async function initNuxt (nuxt: Nuxt) {
   }
 
   // Add <NuxtWelcome>
-  addComponent({
-    name: 'NuxtWelcome',
-    priority: 10, // built-in that we do not expect the user to override
-    filePath: resolve(nuxt.options.appDir, 'components/welcome'),
-  })
+  const islandsConfig = nuxt.options.experimental.componentIslands
+  if (nuxt.options.dev || !islandsConfig || islandsConfig === true || typeof islandsConfig === 'string' || islandsConfig.selectiveClient !== 'deep') {
```

Comment:
Can we make this shorter  ?

Reply:
ideas welcome! - but we also have to type guard given that `componentIslands` can be a string, boolean or object...

Reply:
```suggestion
  if (!(typeof islandConfig === 'object' && islandsConfig.selectiveClient === 'deep') {
```

Reply:
like this ?

Reply:
perfect!